src/main/core/scheduler/mod.rs

```diff
@@ -64,7 +64,7 @@ impl Scheduler {
 }
 
 pub enum SchedulerScope<'sched, 'pool, 'scope> {
-    ThreadPerHost(thread_per_host::SchedulerScope<'pool, 'scope>),
+    ThreadPerHost(thread_per_host::SchedulerScope<'pool, 'scope, Box<Host>>),
     ThreadPerCore(thread_per_core::SchedulerScope<'sched, 'pool, 'scope, Box<Host>>),
```

Comment:
Just curious: why does `ThreadPerCore` need the `'sched` but `ThreadPerHost` does not?

Reply:
The thread-per-core scheduler stores the hosts in queues (which are owned by the scheduler), and the thread-per-host scheduler stores hosts in thread-locals (which have a static lifetime). So the thread-per-host scheduler doesn't need the scheduler lifetime.

```rust
// thread-per-core
pub struct SchedulerScope<'sched, 'pool, 'scope, HostType: Host>
where
    'sched: 'scope,
{
    thread_hosts: &'sched Vec<ArrayQueue<HostType>>,
    ...
```

```rust
// thread-per-host
pub struct SchedulerScope<'pool, 'scope, HostType: Host> {
    ...
    host_storage: &'static LocalKey<RefCell<Option<HostType>>>,
}
```